### PR TITLE
added liquibase and Spring Boot Data JPA to the project

### PR DESCRIPTION
**Pull request by kaiiiseeel**

### CODE REVIEW

File: src/main/java/mate/academy/bookstore/controller/BookController.java [hunk 1]
```diff
@@ -32,4 +34,15 @@ public BookDto getBookById(@PathVariable Long id) {
...
         return bookService.save(requestDto);
     }
+
+    @PutMapping("/{id}")
+    public BookDto updateBookById(@PathVariable Long id,
+                                  @RequestBody CreateBookRequestDto requestDto) {
+        return bookService.update(id, requestDto);
+    }
+
+    @DeleteMapping("/{id}")
```
Comment (karma-o) [inline src/main/java/mate/academy/bookstore/controller/BookController.java:48]: add `no content` response status to here, and `created` for the post endpoint that you already have

File: src/main/java/mate/academy/bookstore/controller/BookController.java [hunk 2]
```diff
@@ -32,4 +34,15 @@ public BookDto getBookById(@PathVariable Long id) {
     public BookDto createBook(@RequestBody CreateBookRequestDto requestDto) {
         return bookService.save(requestDto);
     }
+
+    @PutMapping("/{id}")
+    public BookDto updateBookById(@PathVariable Long id,
+                                  @RequestBody CreateBookRequestDto requestDto) {
```
Comment (karma-o) [inline src/main/java/mate/academy/bookstore/controller/BookController.java]: It isn't just for creation anymore
```suggestion
                                  @RequestBody BookRequestDto requestDto) {

```

File: src/main/java/mate/academy/bookstore/service/impl/BookServiceImpl.java [hunk 3]
```diff
@@ -36,4 +36,19 @@ public BookDto getById(Long id) {
...
+        bookModel.setId(id);
+        return bookMapper.toDto(bookRepository.save(bookModel));
+    }
+
+    @Override
+    public void delete(Long id) {
+        if (bookRepository.existsById(id)) {
+            bookRepository.deleteById(id);
+        }
+        throw new EntityNotFoundException("Can't delete book with such id: " + id);
```
Comment (karma-o) [inline src/main/java/mate/academy/bookstore/service/impl/BookServiceImpl.java]: let's flip this if to follow up with the fail/return early pattern
```suggestion
        if (!bookRepository.existsById(id)) {
            throw new EntityNotFoundException("Can't delete book with such id: " + id);
        }
        bookRepository.deleteById(id);
```

File: src/main/resources/db/changelog/changes/01-create-books-table.yaml [hunk 4]
```diff
@@ -0,0 +1,48 @@
...
+        - createTable:
+            tableName: books
+            columns:
+              - column:
+                  name: id
+                  type: bigint
+                  autoIncrement: true
+                  constraints:
+                    primaryKey: true
+                    nullable: false
```
Comment (karma-o) [inline src/main/resources/db/changelog/changes/01-create-books-table.yaml]: `primaryKey` already has `nullable false` inside it
```suggestion
                    primaryKey: true
```

File: src/main/resources/db/changelog/changes/01-create-books-table.yaml [hunk 5]
```diff
@@ -0,0 +1,48 @@
...
+                  type: varchar(255)
+              - column:
+                  name: cover_image
+                  type: varchar(255)
+              - column:
+                  name: is_deleted
+                  type: boolean
+                  defaultValueBoolean: false
+                  constraints:
+                    nullable: false
```
Comment (karma-o) [inline src/main/resources/db/changelog/changes/01-create-books-table.yaml]:
```suggestion
                    nullable: false

```

File: src/main/resources/db/changelog/db.changelog-master.yaml [hunk 6]
```diff
@@ -0,0 +1,3 @@
+databaseChangeLog:
+  - include:
+      file: db/changelog/changes/01-create-books-table.yaml
```
Comment (karma-o) [inline src/main/resources/db/changelog/db.changelog-master.yaml]:
```suggestion
      file: db/changelog/changes/01-create-books-table.yaml

```

File: src/main/resources/db/changelog/changes/01-create-books-table.yaml [hunk 7]
```diff
@@ -0,0 +1,48 @@
...
+                  name: price
+                  type: decimal(14, 2)
+                  constraints:
+                    nullable: false
+              - column:
+                  name: description
+                  type: varchar(255)
+              - column:
+                  name: cover_image
+                  type: varchar(255)
```
Comment (karma-o) [inline src/main/resources/db/changelog/changes/01-create-books-table.yaml]: I'd make those fields at least 500

**Review by karma-o — CHANGES_REQUESTED** [review]
Looks good so far, left some minor comments

File: src/main/java/mate/academy/bookstore/entity/Book.java [hunk 8]
```diff
@@ -32,4 +36,8 @@ public class Book {
     private BigDecimal price;
     private String description;
     private String coverImage;
+    @Column(nullable = false)
+    private boolean isDeleted = false;
 }
+
+
```
Comment (andrii-hoienko) [inline src/main/java/mate/academy/bookstore/entity/Book.java]:
```suggestion
```

File: src/main/java/mate/academy/bookstore/service/impl/BookServiceImpl.java [hunk 9]
```diff
@@ -36,4 +36,19 @@ public BookDto getById(Long id) {
...
+        Book bookModel = bookMapper.toModel(requestDto);
+        bookModel.setId(id);
+        return bookMapper.toDto(bookRepository.save(bookModel));
+    }
+
+    @Override
+    public void delete(Long id) {
+        if (!bookRepository.existsById(id)) {
+            throw new EntityNotFoundException("Can't delete book with such id: " + id);
+        }
```
Comment (andrii-hoienko) [inline src/main/java/mate/academy/bookstore/service/impl/BookServiceImpl.java:54]: Why you check existence on delete but do not check on update?

File: src/main/resources/application.properties [hunk 10]
```diff
@@ -3,5 +3,6 @@ spring.datasource.url=jdbc:mysql://localhost/book_store?serverTimezone=UTC
 spring.datasource.driverClassName=com.mysql.cj.jdbc.Driver
 spring.datasource.username=
 spring.datasource.password=
-spring.jpa.hibernate.ddl-auto=create-drop
+
```
Comment (andrii-hoienko) [inline src/main/resources/application.properties]:
```suggestion
```

**Review by andrii-hoienko — CHANGES_REQUESTED** [review]
Nice, left minor comments

**Review by karma-o — APPROVED** [review]

**Pull request merged**